src/hls/model/dfcir/include/dfcir/conversions/DFCIRPasses.h

```diff
@@ -0,0 +1,67 @@
+#ifndef DFCIR_PASSES_H
+#define DFCIR_PASSES_H
+
+#include "dfcir/DFCIROperations.h"
+#include "mlir/Pass/Pass.h"
+
+#include "memory"
+
+namespace mlir::dfcir {
+
+enum Ops {
+  UNDEFINED,
+  ADD_INT,
+  ADD_FLOAT,
+  SUB_INT,
+  SUB_FLOAT,
+  MUL_INT,
+  MUL_FLOAT,
+  DIV_INT,
+  DIV_FLOAT,
+  AND_INT,
+  AND_FLOAT,
+  OR_INT,
+  OR_FLOAT,
+  XOR_INT,
+  XOR_FLOAT,
+  NOT_INT,
+  NOT_FLOAT,
+  NEG_INT,
+  NEG_FLOAT,
+  LESS_INT,
+  LESS_FLOAT,
+  LESS_EQ_INT,
+  LESS_EQ_FLOAT,
+  MORE_INT,
+  MORE_FLOAT,
+  MORE_EQ_INT,
+  MORE_EQ_FLOAT,
+  EQ_INT,
+  EQ_FLOAT,
+  NEQ_INT,
+  NEQ_FLOAT
+};
+
+} // namespace mlir::dfcir
+
+typedef std::unordered_map<mlir::dfcir::Ops, unsigned> LatencyConfig;
+
+
+namespace mlir::dfcir {
+
+using std::unique_ptr;
+using mlir::Pass;
+
+unique_ptr<Pass> createDFCIRToFIRRTLPass(LatencyConfig *config = nullptr);
+
+unique_ptr<Pass> createDFCIRDijkstraSchedulerPass();
+
+unique_ptr<Pass> createDFCIRLinearSchedulerPass();
+
+} // namespace mlir::dfcir
+
+#define GEN_PASS_REGISTRATION
+
+#include "dfcir/conversions/DFCIRPasses.h.inc"
```

Comment:
These are the `*.inc`-headers generated by TableGen. They are needed to reduce the boilerplate code which otherwise would have been written by hand. They are also the reason why I cannot enforce sorting the list of included files (they refer to the functionality declared in other included files, breaking the sorted sequence).